geode-core/src/main/java/org/apache/geode/cache/wan/GatewaySender.java

```diff
@@ -215,6 +216,34 @@ enum OrderPolicy {
    */
   void startWithCleanQueue();
 
+
+  /**
+   * Prepare GatewaySender for closing of Cache.
+   *
+   * <p>
+   * Implementation of new API in ParallelGatewaySenderImpl:
+   *
+   *
+   * <pre>
+   * public void prepareForStop() {
+   *   if (!isRunning()) {
+   *     return;
+   *   }
+   *   pause();
+   *   if (eventProcessor != null &amp;&amp; !eventProcessor.isStopped()) {
+   *     eventProcessor.prepareForStopProcessing();
+   *   }
+   * }
+   * </pre>
+   *
+   *
+   * <p>
+   * Invoked at closing of cache.
+   *
+   */
+  @Experimental
+  void prepareForStop();
```

Comment:
All experimental APIs should have javadoc and annotation explanation strings. Please see [MetricsPublishingService ](https://github.com/apache/geode/blob/develop/geode-core/src/main/java/org/apache/geode/metrics/MetricsPublishingService.java) as an example.

Reply:
Since changes are only for internal use, impacts moved to InternalGatewaySender interface.

Reply:
Hi @kirklund , are changes OK with you?

Reply:
Hi @kirklund , are changes now OK?